### PR TITLE
added Design Patterns task

### PR DESCRIPTION
**Pull request by irynayeutushenka**

### CODE REVIEW

File: src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    @Test(description = "Successful swapping of award programs for 5021 client", priority = 0)
+    public void swapAwardProgramsTest() throws InterruptedException {
+        loginToIaf(getWebDriver(), iafConsoleUser);
+        new IafHomePage(getWebDriver())
+                .openConfigApp()
+                .clickClientLink(CLIENT_5021.getConfigAppName())
+                .moveToAwardPrograms();
+
+        String textInitial = getWebDriver().findElement
+                (By.xpath("//tbody[@class='yui-dt-data']/tr[1]//a")).getText();
```
Comment (aliaksandrazaronak) [inline src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java]: We should avoid of findElement() usage within test methods. Please create appropriate method for corresponding page and re-use it in needed tests

File: src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        System.out.println("1st program BEFORE movement is " + textInitial);
+
+        if (textInitial.contains("testclient5021 program")) {
+            new AwardProgramsPage(getWebDriver())
+                    .moveAwardProgram();
+
+            String textUpdated = getWebDriver().findElement
+                    (By.xpath("//tbody[@class='yui-dt-data']/tr[1]//a")).getText();
+
```
Comment (aliaksandrazaronak) [inline src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java]: The same thing that I mentioned above

File: src/test/java/tests/payroll_job/LoginToSchedulerConsoleTest.java [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+
 public class LoginToSchedulerConsoleTest extends BaseTest {
 
     private final String pageName = "Scheduler Console";
 
-    private SchedulerConsoleUser user = new SchedulerConsoleUser();
+    UserBO schedulerConsoleUser = UserFactory.createSchedulerConsoleUser();
 
     @Test(description = "Successful login to Scheduler Console", priority = 0,
             groups = {"loginToSchedulerConsole", "payroll_test"})
```
Comment (aliaksandrazaronak) [inline src/test/java/tests/payroll_job/LoginToSchedulerConsoleTest.java]: Priority equals 0 is by default. You can remove it from current test method.

File: src/test/java/tests/common_test/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    @Parameters({"browser"})
+
+    @BeforeClass
+    public void beforeTest(@Optional("chrome") String browser) {
+        driverManager = DriverManagerFactory.getManager(browser);
+    }
+
+    @BeforeMethod
+    public void beforeMethod() {
+        webDriver = driverManager.getInstance();
```
Comment (aliaksandrazaronak) [inline src/test/java/tests/common_test/BaseTest.java]: Why can't we get webDriver instance within @BeforeClass?

File: src/test/java/tests/common_test/BaseTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    @AfterMethod
+    public void closeWebDriver() {
+        if (webDriver != null) {
+            webDriver.quit();
+        }
```
Comment (aliaksandrazaronak) [inline src/test/java/tests/common_test/BaseTest.java]: webDriver.quit() is more convenient and universal.

File: src/test/java/tests/common_test/BaseTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        webDriver = null;
+    }
```
Comment (aliaksandrazaronak) [inline src/test/java/tests/common_test/BaseTest.java]: We don't need to set webDriver = null after executing webDriver.close() - java has Garbage Collector, which clean up all unnecessary instances.

**Review by aliaksandrazaronak — CHANGES_REQUESTED** [review]

File: src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java (hunk 1 above)
Comment (irynayeutushenka) [reply src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java]: Fixed

File: src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java (hunk 2 above)
Comment (irynayeutushenka) [reply src/test/java/tests/config_app_tests/award_programs_tests/SwapAwardProgramsTest.java]: Fixed

File: src/test/java/tests/payroll_job/LoginToSchedulerConsoleTest.java (hunk 3 above)
Comment (irynayeutushenka) [reply src/test/java/tests/payroll_job/LoginToSchedulerConsoleTest.java]: Fixed

File: src/test/java/tests/common_test/BaseTest.java (hunk 4 above)
Comment (irynayeutushenka) [reply src/test/java/tests/common_test/BaseTest.java]: Fixed

File: src/test/java/tests/common_test/BaseTest.java (hunk 5 above)
Comment (irynayeutushenka) [reply src/test/java/tests/common_test/BaseTest.java]: Renamed

File: src/test/java/tests/common_test/BaseTest.java (hunk 6 above)
Comment (irynayeutushenka) [reply src/test/java/tests/common_test/BaseTest.java]: Fixed

File: src/test/java/tests/common_test/BaseTest.java (hunk 4 above)
Comment (aliaksandrazaronak) [reply src/test/java/tests/common_test/BaseTest.java]: Approved

File: src/test/java/tests/common_test/BaseTest.java (hunk 5 above)
Comment (aliaksandrazaronak) [reply src/test/java/tests/common_test/BaseTest.java]: Approved

File: src/test/java/tests/common_test/BaseTest.java (hunk 6 above)
Comment (aliaksandrazaronak) [reply src/test/java/tests/common_test/BaseTest.java]: Approved

File: src/test/java/tests/payroll_job/LoginToSchedulerConsoleTest.java (hunk 3 above)
Comment (aliaksandrazaronak) [reply src/test/java/tests/payroll_job/LoginToSchedulerConsoleTest.java]: Approved

**Review by aliaksandrazaronak — CHANGES_REQUESTED** [review]

**Pull request merged**